src/main/scala/uk/gov/hmrc/zap/api/ZapAlerts.scala

```diff
@@ -41,9 +41,19 @@ class ZapAlerts(zapClient: ZapClient) {
       relevantAlerts
   }
 
-
   def parsedAlerts: List[ZapAlert] = {
-    val response: String = callZapApi("/json/core/view/alerts", "baseurl" -> alertsBaseUrl)
+    var alerts: List[ZapAlert] = List.empty
```

Comment:
In Scala, an `if` is actually an expression like any other block of code, which means you can directly use it as a `return` value. The below is valid syntax.

```scala
val a: Int = if (true) 5 else 10
```

So what you could do:

```scala
def parsedAlerts: List[ZapAlert] = {
  if (alertUrlsToReport.isEmpty) {
    getAlerts()
  } else {
     getAlerts(alertUrl) ::: alertUrlsToReport 
  }
}
```

Even better, usually when you deal with `List` in scala, you can use pattern matching, to have cleaner syntax:

```scala
def parsedAlerts: List[ZapAlert] = {
  alertUrlsToReport match {
    case Nil => getAlerts()
    case head :: tail => getAlerts(alertUrl) ::: alertUrlsToReport
  }
}
```

Hopefully I understood the condition right.

Reply:
Changes incorporated.